### PR TITLE
Protect lookup in zone index to avoid crash in debug

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
Array index crash accidentally dropped into develop with #9958.  This fixes that crash by protecting the array lookup until it's actually needed.  FYI @jmarrec @mjwitte 

If this passes the integration debug tests, it will merge.  (There will still be one unrelated debug failure due to SSC (#9984))

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -10991,15 +10991,15 @@ void WriteVeriSumTable(EnergyPlusData &state)
                     curArea += frameArea;
                 }
                 int const zonePt = thisSurf.Zone;
-                auto const &thisZone = state.dataHeatBal->Zone(zonePt);
 
-                bool const isConditioned = (zonePt > 0) && (thisZone.SystemZoneNodeNumber > 0);
+                bool const isConditioned = (zonePt > 0) && (state.dataHeatBal->Zone(zonePt).SystemZoneNodeNumber > 0);
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabular.cc:10995]: This boolean short circuit works fine as long as you defer array lookup until you know it is valid.

File: src/EnergyPlus/OutputReportTabular.cc [hunk 2]
```diff
@@ -10991,15 +10991,15 @@ void WriteVeriSumTable(EnergyPlusData &state)
...
                 if ((thisSurf.Tilt >= 60.0) && (thisSurf.Tilt <= 120.0)) {
                     // vertical walls and windows
                     switch (thisSurf.Class) {
                     case SurfaceClass::Wall:
                     case SurfaceClass::Floor:
                     case SurfaceClass::Roof: {
+                        auto const &thisZone = state.dataHeatBal->Zone(zonePt);
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabular.cc:11002]: Wait until the blocks where we know we have a zone to do the array lookup.

**Review by Myoldmopar — COMMENTED** [review]
Minimal changes, minimal walkthrough.

File: src/EnergyPlus/OutputReportTabular.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/OutputReportTabular.cc:10995]: Same type of CppCheck change error I made in HVACManager.

**Comment by Myoldmopar** [comment]
OK this is good on CI.  Still one failing test file, but not two, so merging this.

**Pull request merged**